pkgs/stdenv/generic/setup.sh

```diff
@@ -68,9 +68,10 @@ _callImplicitHook() {
     local hookName="$2"
     if declare -F "$hookName" > /dev/null; then
         "$hookName"
-    elif type -p "$hookName" > /dev/null; then
```

Comment:
This is not semantically equivalent.

* `type -p` looks up the PATH for a program called "$hookName". If it exists, it outputs its location and exits with 0. Otherwise it exits with 1.
* `test -e` checks if a file located at "$hookName" exists. If it does, it exits with 0. Otherwise it exits with 1.

Reply:
There is a case where both would yield the same result, if `$hookName` points to a relative *and* executable path.

Reply:
Thank you for the feeback!
I think you're right here and that change is a mistake.
I'm going to do one more test, but it seems weird to me that the original code is generating an error. From what I understood, in bash, if a command is in an `if` statement then the errexit should be ignored.
Perhaps the strict_all flag of osh is removing that behavior.
I'll test again and will ask andrew if that is the case.